ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java

```diff
@@ -122,9 +122,9 @@ public void onEth2GenesisEvent(Eth2GenesisEvent event) {
   }
 
   @Subscribe
-  public void onDeposit(Deposit deposit) {
+  public void onDeposit(tech.pegasys.artemis.pow.event.Deposit event) {
     if (deposits == null) deposits = new ArrayList<Deposit>();
-    deposits.add(deposit);
+    deposits.add(DepositUtil.convertEventDepositToOperationDeposit(event));
```

Comment:
This is very cool. Thanks for separating client-specific data structure and spec data structure in such a visible way. 💯